python/foglamp/services/core/api/south.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+from aiohttp import web
+
+from foglamp.common.service_record import ServiceRecord
+from foglamp.common.storage_client.payload_builder import PayloadBuilder
+from foglamp.services.core.service_registry.service_registry import ServiceRegistry
+from foglamp.services.core.service_registry.exceptions import DoesNotExist
+from foglamp.services.core import connect
+
+
+__author__ = "Praveen Garg"
+__copyright__ = "Copyright (c) 2018 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_help = """
+    -------------------------------------------------------------------------------
+    | GET                 | /foglamp/south                                        |
+    -------------------------------------------------------------------------------
+"""
+
+
+async def _get_tracked_assets_and_readings(storage_client, svc_name):
+    asset_json = []
+    payload = PayloadBuilder().SELECT("asset").WHERE(['service', '=', svc_name]).payload()
+    try:
+        result = await storage_client.query_tbl_with_payload('asset_tracker', payload)
+        asset_records = result['rows']
+
+        _readings_client = connect.get_readings_async()
+        for r in asset_records:
+            payload = PayloadBuilder().AGGREGATE(["count", "*"]).ALIAS("aggregate", ("*", "count", "count")) \
+                .GROUP_BY("asset_code").WHERE(['asset_code', '=', r["asset"]]).payload()
+            results = await _readings_client.query(payload)
+            response = results['rows']
+            asset_json = [{"count": r['count'], "asset": r['asset_code']} for r in response]
+    except:
+        raise
+    else:
+        return asset_json
+
+
+async def _get_southbound_service_records():
+    storage_client = connect.get_storage_async()
+    sr_list = list()
+    try:
+        svc_records = ServiceRegistry.get(s_type="Southbound")
+    except DoesNotExist:
+        pass
+    else:
+        for service_record in svc_records:
+            sr_list.append(
+                {
+                    'name': service_record._name,
+                    'address': service_record._address,
+                    'management_port': service_record._management_port,
+                    'service_port': service_record._port,
+                    'protocol': service_record._protocol,
+                    'status': ServiceRecord.Status(int(service_record._status)).name.lower(),
+                    'assets': await _get_tracked_assets_and_readings(storage_client, service_record._name)
```

Comment:
Why limit this to those that are currently registered? We may have buffered readings for south services that are not currently running.

Reply:
I can drive it that way, 1) fetch all services & asset, fetch corresponding readings 2) fetch services from servicesRegistry and merge that info 1 as applicable.

In this case, we will have no info for ports, protocol, status.

Reply:
As discussed and with FOGL-1701, we should be good with it for now. Or we want to get old services as well, which were b/w restart.

Reply:
@MarkRiddoch  addressed in https://github.com/foglamp/FogLAMP/pull/1029